### PR TITLE
Allow creating windows initially disabled

### PR DESCRIPTION
**Pull request by vadz**
I've found a long-standing [ticket #16385](http://trac.wxwidgets.org/ticket/16385) and decided to finally fix it.

Under MSW this actually looks pretty simple and I'm not sure why hadn't we done this a long time ago (@catalinr am I missing/forgetting something here?), but under GTK this is trickier as several classes override `Enable()` and so calling `DoEnable(false)` after creating a disabled control wouldn't work there and I had to switch from overriding the former to overriding the latter. I _think_ this should be fine and things seem to work when testing in the widgets sample (basically just pressing `Ctrl-E` on all the relevant pages), but I'd appreciate a confirmation from Paul.

In any case, I'm not going to commit this just before the release, but if you don't see anything wrong with it, I'll commit it soon afterwards.

### CODE REVIEW

File: src/gtk/spinbutt.cpp [hunk 1]
```diff
@@ -173,16 +173,16 @@ void wxSpinButton::SetRange(int minVal, int maxVal)
...
 
-bool wxSpinButton::Enable( bool enable )
+void wxSpinButton::DoEnable(bool enable)
 {
-    if (!base_type::Enable(enable))
-        return false;
+    if ( !m_widget )
+        return;
+
+    base_type::Enable(enable);
```
Comment (paulcor) [inline src/gtk/spinbutt.cpp]: Shouldn't this be `DoEnable()`?

**Comment by MaartenBent** [comment]
So this segfaults in AppVeyor. Not sure why but my best guess is you should not use `delete m_button` but `m_button->Destroy()`. Otherwise the main window does not know it is deleted when it iterates over all children after calling `Disable()`.

**Comment by vadz** [comment]
Oops. I don't really see the problem because destroying the window calls `m_parent->RemoveChild(this)` from its dtor, so there should be no dangling pointers remaining here, but I do see the crash :-( I'll rebuild with MinGW and will try to debug it locally, thanks.

**Comment by vadz** [comment]
Well, isn't that great. The test suite passes perfectly fine for me locally:

```
wx\HEAD\tests>mgw81_mswud\test_gui.exe
Test program for wxWidgets GUI features
build: 3.1.3 (wchar_t,compiler with C++ ABI compatible with gcc 4,wx containers,compatible with 3.0)
running under Windows 7 (build 7601, Service Pack 1), 64-bit edition as zeitlin, locale is C
===============================================================================
All tests passed (7825 assertions in 110 test cases)
```

So either it's specific to MSW 10 or gcc 5.3.

**Pull request merged**